### PR TITLE
Use in-class member initializers

### PR DESCRIPTION
**Pull request by ChrisThrasher**
## Description

Continuing the work from #2156 and #2275.

### CODE REVIEW

**Comment by eXpl0it3r** [comment]
SFMLBuildMaster: Build this please.

File: include/SFML/Graphics/Text.hpp [hunk 1]
```diff
@@ -428,20 +428,20 @@ class SFML_GRAPHICS_API Text : public Drawable, public Transformable
...
-    mutable VertexArray   m_vertices;            //!< Vertex array containing the fill geometry
-    mutable VertexArray   m_outlineVertices;     //!< Vertex array containing the outline geometry
-    mutable FloatRect     m_bounds;              //!< Bounding rectangle of the text (in local coordinates)
-    mutable bool          m_geometryNeedUpdate;  //!< Does the geometry need to be recomputed?
-    mutable std::uint64_t m_fontTextureId;       //!< The font texture id
+    String                m_string;                                    //!< String to display
+    const Font*           m_font{nullptr};                             //!< Font used to display the string
+    unsigned int          m_characterSize{30};                         //!< Base size of characters, in pixels
+    float                 m_letterSpacingFactor{1};                    //!< Spacing factor between letters
+    float                 m_lineSpacingFactor{1};                      //!< Spacing factor between lines
```
Comment (eXpl0it3r) [inline include/SFML/Graphics/Text.hpp:435]: I'd use `1.f`

File: include/SFML/Graphics/Transform.hpp [hunk 2]
```diff
@@ -266,7 +266,12 @@ class Transform
     ////////////////////////////////////////////////////////////
     // Member data
     ////////////////////////////////////////////////////////////
-    float m_matrix[16]; //!< 4x4 matrix defining the transformation
+    // clang-format off
+    float m_matrix[16]{1, 0, 0, 0,
+                       0, 1, 0, 0,
+                       0, 0, 1, 0,
+                       0, 0, 0, 1}; //!< 4x4 matrix defining the transformation
```
Comment (eXpl0it3r) [inline include/SFML/Graphics/Transform.hpp:273]: I'd keep the `.f` suffix

**Review by eXpl0it3r — CHANGES_REQUESTED** [review]
I'd keep the literals, otherwise it looks okay

File: include/SFML/Graphics/Text.hpp (hunk 1 above)
Comment (ChrisThrasher) [reply include/SFML/Graphics/Text.hpp:435]: I can do that, but it's worth remembering that using `{}` ensures at compile time that no data loss is occurring.

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Review by eXpl0it3r — APPROVED** [review]

File: include/SFML/Graphics/Shape.hpp [hunk 3]
```diff
@@ -304,15 +304,15 @@ class SFML_GRAPHICS_API Shape : public Drawable, public Transformable
...
-    FloatRect      m_bounds;           //!< Bounding rectangle of the whole shape (outline + fill)
+    const Texture* m_texture{nullptr};           //!< Texture of the shape
+    IntRect        m_textureRect;                //!< Rectangle defining the area of the source texture to display
+    Color          m_fillColor{Color::White};    //!< Fill color
+    Color          m_outlineColor{Color::White}; //!< Outline color
+    float          m_outlineThickness{0};        //!< Thickness of the shape's outline
+    VertexArray    m_vertices{PrimitiveType::TriangleFan};          //!< Vertex array containing the fill geometry
+    VertexArray    m_outlineVertices{PrimitiveType::TriangleStrip}; //!< Vertex array containing the outline geometry
+    FloatRect      m_insideBounds;                                  //!< Bounding rectangle of the inside (fill)
+    FloatRect      m_bounds; //!< Bounding rectangle of the whole shape (outline + fill)
```
Comment (Jim-Marsden) [inline include/SFML/Graphics/Shape.hpp:315]: I don't know if it's an issue, but on m_bounds, is there consistent indention for the comments?

Comment (ChrisThrasher) [reply include/SFML/Graphics/Shape.hpp:315]: This is a quirk of clang-format trying to maintain vertical alignment with exceeding out specified line width. Not much we can do about it without disabling certain vertical alignment rules across the codebase.

**Pull request merged**